pkg/service/model_builder.go

```diff
@@ -214,3 +221,19 @@ func (t *defaultModelBuildTask) getDeletionProtectionViaAnnotation(svc corev1.Se
 	}
 	return false, nil
 }
+
+func (t *defaultModelBuildTask) needsCleanup() bool {
+	if !t.service.DeletionTimestamp.IsZero() {
+		return true
+	}
+	if t.service.Spec.Type != corev1.ServiceTypeLoadBalancer {
+		if t.featureGates.Enabled(config.ServiceTypeLoadBalancerOnly) {
+			return true
+		}
+		var lbType string
+		if !t.annotationParser.ParseStringAnnotation(annotations.SvcLBSuffixLoadBalancerType, &lbType, t.service.Annotations) {
```

Comment:
how about user changed it to something else like "nlb-ddd", seems we won't delete the NLB here.

Reply:
also, how about other cases like when the nlb-target-type annotation is removed while lb-type is still external (when type changed back to node port).

I feel such logic should be in a centralized place(shared by both event handler and here)

Reply:
our docs mention the aws-load-balancer-type annotation should be treated immutable. We can definitely cleanup the NLB if we no longer recognize the annotation.

I agree, we should centralize the logic to determine whether the service is supported or not.